WordPress/src/main/java/org/wordpress/android/AppInitializer.kt

```diff
@@ -1019,8 +1020,9 @@ class AppInitializer @Inject constructor(
         /**
          * Update locale of the static context when language is changed.
          */
-        fun updateContextLocale() {
-            context = LocaleManager.setLocale(context)
+        fun updateContextLocale(appContext: Context? = null) {
+            val context = appContext ?: context!!
```

Comment:
Question (❓): Why were the double exclamation marks (`!!`) added to the `context`? Is that by design, do we want the `updateContextLocale(...)` function to crash exactly there, that is, when the `context` is not already initialized in advance? 🤔

Reply:
tbh I have no clue about this one, you'll have to ask @ovitrif as the author of this commit 😅 

The reason for this change on `updateContextLocale` is that when we tried to run the UI tests before that change, the `TestRules` that we have on the screenshot tests tried to change the context before the `AppInitializer.kt` singleton was initialized (as the `TestRule` where `@ClassRule` and executed way before all that) which led to a NPE (see the first checkbox with a 🕵️ in my PR description), and which is the reason why we had to change the code to pass the context to `updateContextLocale` here. But know being a Kotlin dev, I have no idea if that `!!` vs `!` vs other solutions was the right one for this implementation of the fix.

Reply:
> Question (❓): Why were the double exclamation marks (!!) added to the context? Is that by design, do we want the updateContextLocale(...) function to crash exactly there, that is, when the context is not already initialized in advance? 🤔

Good thinking @ParaskP7!

Initially investigation into the issue described by @AliSoftware started with an error thrown somewhere deep in the code when a "random" (not easily leading to the root cause of the issue) method was getting called, the only relevant clue we got was that the `context` was null but this wasn't very clear eiter.

To improve things, I updated the code to fail fast, and as a result of more refactoring I've ended up with this code to make sure we're avoiding such a case also in this place.

Reply:
Oh cool, thanks so much for the extra clarification on that @ovitrif ! 🙇 

Can I maybe suggest a follow-up on that then, maybe adding a comment or some kind of documentation on that, that is, in case we anyway proceed with keeping the `WPLocaleTestRule` test rule. I recommend this action just so that to make sure we avoid this kind of question in the future, or avoid having someone like me, when looking as that, to be thinking, oh, but this `!!` is unnecessary, and ending up removing it silently anyway. Wdyt? 🤔

Reply:
Sounds good to me Petros 👍

Reply:
@ParaskP7 While I mainly agree that maybe we should remove `WPLocaleTestRule` after all (as discussed in the other comment thread of this PR), what will likely happen is that I'll keep it in this PR for now because I want to avoid making another big change on this PR (like removing `WPLocaleTestRule`) if possible so that it doesn't risk delaying the landing of that PR (since more follow-up work, including some work to fix Android issues like #17128, depend on this to land, so I don't want to block those either).

So my plan is:
 - Let Ovi fix address your suggestion on adding the adequate code comment on the `!!`
 - Hopefully find someone with a M1 (and ideally someone who doesn't work on Android daily and thus might not have the necessary tools and system images installed on their Mac, to ensure the lane errors correctly and installs the necessary things when they are not installed yet…) to add an additional review to this PR… though if no one manifests I'll likely end up merging it anyway to move forward
 - Consider removing `WPLocaleTestRule` in a follow-up PR. Which means at this point the changes around `updateContextLocale` now taking `context` as a parameter instead of infering it from local value would likely stay even when I'll remove `WPLocaleTestRule`… and I think that's not a bad thing (Dependency Injection over Singleton and inner hardcoded values ftw) if we keep those changes made by Ovi anyway even if the TestRule disappears?

Reply:
Sounds like a plan to me, thanks for the extra clarification on keep the `WPLocaleTestRule` @AliSoftware , much appreciated! 🥇

> ...and I think that's not a bad thing if we keep those changes made by Ovi anyway even if the TestRule disappears?

I am not sure on that, but I don't have a strong opinion on it too. I am always in favor of less complexity, less code, less code related documentation (when possible). Thus, if you feel that even after removing this `WPLocaleTestRule` we might still benefit from all that, then I wouldn't debate on it again. All I wanted to do is becoming the devil's advocate and make you double think this update. 😈

Reply:
Added method docs + contract check and error message via this suggestion PR: https://github.com/wordpress-mobile/WordPress-Android/pull/17196 👍